pkg/autoscaler/autoscaler.go

```diff
@@ -106,9 +121,34 @@ func (agg *perPodAggregation) aggregate(concurrency float64) {
 	agg.probeCount += 1
 }
 
+func (agg *perPodAggregation) lameduck(t *time.Time) {
+	if agg.lameduckTime == nil {
+		agg.lameduckTime = t
+	}
+	if agg.lameduckTime.After(*t) {
+		agg.lameduckTime = t
+	}
+}
+
 // Calculates the average concurrency over all values given
-func (agg *perPodAggregation) calculateAverage() float64 {
-	return agg.accumulatedConcurrency / float64(agg.probeCount)
+func (agg *perPodAggregation) calculateAverage(now time.Time) float64 {
+	if agg.probeCount == 0 {
+		return 0.0
+	}
+	avg := agg.accumulatedConcurrency / float64(agg.probeCount)
+	if agg.lameduckTime == nil {
+		return avg
+	}
+	return agg.weightedPodCount(now) * avg
```

Comment:
I think we can safely drop the extra if here and return `agg.accumulatedConcurrency / float64(agg.probeCount) * agg.weightedPodCount(now)`. `weightedPodCount` short circuits the `if lameduck == nil` as well and it makes for a little less complexity here.

Reply:
Yes, that simpler.  Done.